### ISSUE TITLE
Detect whether screen is locked / on

### ISSUE THREAD

**Issue by beyang** [sourcegraph/thyme#1]

**Comment by mintbridge** [comment]
yes please!, Im not sure about erasing it but it would be nice if the "active" window was set to locked or something similar when the screen is locked or in standby etc

**Comment by vltr** [comment]
I agree with @mintbridge . Erase is not something you want, but at least to know when you had your screen locked would be perfect to a better time tracking. Perhaps add it as an active application would be nothing but the truth. Meanwhile, someone can say: "hey, I was cross compiling an entire toolchain for [insert your arch here] and that counts". Agreed too, so a new line on top of the active windows showing when the screensaver was on? I don't know yet about inactivity, but probably this would be secondary IMHO. Just my two cents ... :wink:

**Comment by antonmry** [comment]
No the best solution out there but this script I'm using works in Linux: https://gist.github.com/antonmry/a54b04f6a3536b98924255ae9fbe71d3

Basically, it opens a Terminal with the window name Absent. Pressing q, you close it. As I said, not the best solution but it may be helpful for someone while we have the final solution

**Comment by vltr** [comment]
I don't think it would be too hard to achieve this (keep reading and don't throw any rocks please), *but* I just can say that because I use linux, I'm under xorg, thyme is running as a custom systemd.service triggered by a custom systemd.timer in my userspace and I can create a condition (using `qdbus`) to call thyme, wrapped as a script:

```sh
if [ $(qdbus --session org.gnome.ScreenSaver /org/gnome/ScreenSaver org.gnome.ScreenSaver.GetActive) = 'false' ]; then
  /home/richard/work/thyme/bin/thyme track -o "/home/richard/storage/.daily-logs/data/thyme.json.$(date +%Y%m%d)"
fi
```

But, as I said, this solutions works _for my box / scenario_. Anyway, I think that `dbus` may be a way of getting and/or triggering scripts related to the screensaver. But that's not even half the problem ...

**Comment by vltr** [comment]
Update: I just find out that systemd now comes with a dbus tool, `busctl`. In this case, I had to change my script to `busctl --user call org.gnome.ScreenSaver /org/gnome/ScreenSaver org.gnome.ScreenSaver GetActive`
